src/components/activities-test.js

```diff
@@ -40,78 +45,212 @@ describes.realWin('ActivityPorts test', {}, env => {
 
   afterEach(() => {});
 
-  describe('test delegation', () => {
-    it('should delegate openIframe to ActivityIframePort', () => {
-      const activityPorts = new ActivityPorts(win);
-      sandbox
-        .stub(ActivityIframePort.prototype, 'connect')
-        .callsFake(() => Promise.resolve());
-      return activityPorts.openIframe(iframe, url).then(port => {
-        expect(port instanceof ActivityIframePort).to.be.true;
-      });
+  describe('ActivityPorts', () => {
+    let deps;
+    let activityPorts;
+    let analytics;
+    let pageConfig;
+
+    beforeEach(() => {
+      pageConfig = new PageConfig(publicationId, false);
+      deps = {
+        win: () => win,
+        pageConfig: () => pageConfig,
+        doc: () => doc,
+        eventManager: () => new ClientEventManager(Promise.resolve()),
+      };
+      activityPorts = new ActivityPorts(deps);
+      deps['activities'] = () => activityPorts;
+      analytics = new AnalyticsService(deps);
+      deps['analytics'] = () => analytics;
     });
 
-    it('should delegate open', () => {
-      const activityPorts = new ActivityPorts(win);
-      sandbox.stub(WebActivityPorts.prototype, 'open').callsFake(() => {
-        return {targetWin: null};
+    describe('default arguments', () => {
+      let expectedDefaults;
+
+      beforeEach(() => {
+        expectedDefaults = {
+          'analyticsContext': analytics.getContext().toArray(),
+          'publicationId': pageConfig.getPublicationId(),
+          'productId': pageConfig.getProductId(),
+          '_client': 'SwG $internalRuntimeVersion$',
+        };
       });
-      const opener = activityPorts.open(
-        'some_request_id',
-        '/someUrl',
-        '_top',
-        {}
-      );
-      expect(opener.targetWin).to.be.null;
-    });
 
-    it('must delegate onResult', () => {
-      const activityPorts = new ActivityPorts(win);
-      const resultHandler = portDef => {
-        return portDef.acceptResult().then(result => {
-          expect(result.data).to.deep.equal('test');
+      it('should accept no arguments', () => {
+        expect(activityPorts.addDefaultArguments()).to.deep.equal(
+          expectedDefaults
+        );
+      });
+
+      it('should fill in an empty object', () => {
+        expect(activityPorts.addDefaultArguments({})).to.deep.equal(
+          expectedDefaults
+        );
+      });
+
+      it('should add to an existing object', () => {
+        const newArgs = {
+          aVal: 1,
+        };
+        expectedDefaults['aVal'] = 1;
```

Comment:
Instead of modifying expectedDefaults, could you create a new object within this test using Object.assign?

Reply:
Or maybe `expectedDefaults` could be renamed to `expectedObject`